src/script/ismine.cpp

```diff
@@ -42,17 +50,18 @@ bool HaveKeys(const std::vector<valtype>& pubkeys, const CKeyStore& keystore)
     return true;
 }
 
-isminetype IsMineInner(const CKeyStore& keystore, const CScript& scriptPubKey, bool& isInvalid, IsMineSigVersion sigversion)
+void Update(IsMineResult& val, IsMineResult update)
 {
-    isInvalid = false;
+    if (update > val) val = update;
```

Comment:
I think the name `Update` is unclear, though I have a hard time thinking of a different term. Effectively it combines two `IsMineResults` and chooses the value with the highest "ownership" (apart from INVALID). Don't know if this would work, or be a good fit:
```c++
Update(ret, IsMineResult::WATCH_ONLY);
```
to
```c++
ret = std::max(ret, IsMineResult::WATCH_ONLY);
```

Reply:
Done. The `Update` function was introduced before @jimpo noticed it was really a simple maximum if INVALID is made the highest value.